module/VuFind/src/VuFind/ILS/Driver/MultiBackend.php

```diff
@@ -190,20 +190,53 @@ public function getStatus($id)
      */
     public function getStatuses($ids)
     {
-        $items = [];
+        // Group records by source and request statuses from the drivers
+        $grouped = [];
         foreach ($ids as $id) {
-            try {
-                $items[] = $this->getStatus($id);
-            } catch (ILSException $e) {
-                $items[] = [
-                    [
-                        'id' => $id,
-                        'error' => 'An error has occurred'
-                    ]
+            $source = $this->getSource($id);
+            if (!isset($grouped[$source])) {
+                $driver = $this->getDriver($source);
+                $grouped[$source] = [
+                    'driver' => $driver,
+                    'ids' => []
                 ];
             }
+            $grouped[$source]['ids'][] = $id;
+        }
+
+        // Process each group
+        $results = [];
+        foreach ($grouped as $source => $current) {
+            if ($current['driver']) {
+                $localIds = array_map(
+                    function ($id) {
+                        return $this->getLocalId($id);
+                    },
+                    $current['ids']
+                );
+                try {
+                    $statuses = $current['driver']->getStatuses($localIds);
+                } catch (ILSException $e) {
+                    $statuses = [];
+                    foreach ($localIds as $id) {
+                        $statuses[] = [
+                            [
+                                'id' => $id,
+                                'error' => 'An error has occurred'
+                            ]
+                        ];
+                    }
+                }
+                $statuses = array_map(
+                    function ($status) use ($source) {
+                        return $this->addIdPrefixes($status, $source);
```

Comment:
I haven't taken a very close look at the code yet, but this stands out: I'm pretty sure you can't call `$this` in this context. Have you tested any cases that lead to this code?

Reply:
I have and it works, but I'll verify and change it if it's something that should not be done.

Reply:
Ok, so this should be fine since PHP 5.4 as $this is automatically bound (see http://php.net/manual/en/functions.anonymous.php#example-167).

Reply:
Thanks for double-checking that. I'd swear that I had trouble with something similar to this in the past, but it's just as likely that I simply assumed it wouldn't work and never tried. In any case, I've learned something. I'll review the rest of the code more closely in the near future.